### DISCUSSION TITLE
Could a onnx model use several gpu devices to accelerate?

### DISCUSSION THREAD

**Discussion by posa88** [microsoft/onnxruntime#10527] [Other Q&A]
`
  optionsOfSession1.addCuda(0);
  optionsOfSession1.addCuda(1);

  optionsOfSession2.addCuda(3);
  optionsOfSession2.addCuda(4);
`
Could the code works fine?

**Comment by snnn** [comment]
No. They need be in different processes. 

Our CUDA EP has a threadlocal cache: https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cuda/cuda_execution_provider.h#L113 . Each PerThreadContext maps to one single GPU.
